modules/transport/Transport.js

```diff
@@ -0,0 +1,191 @@
+import EventEmitter from 'events';
+
+import {
+    MESSAGE_TYPE_EVENT,
+    MESSAGE_TYPE_RESPONSE,
+    MESSAGE_TYPE_REQUEST
+} from './constants';
+import PostMessageTransport from './PostMessageTransport';
+
+/**
+ * Stores the currnet transport that have to be used.
+ */
+export default class Transport extends EventEmitter {
+    /**
+     * Creates new instance.
+     *
+     * @param {Object} options - Optional parameters for configuration of the
+     * transport.
+     */
+    constructor(options = {}) {
+        super();
+
+        const { defaultTransportOptions, noDefaultTransport, transport }
+            = options;
+
+        // We need this because addListener in overriden
+        this.on = this.addListener;
+
+        this._requestID = 0;
+
+        if (transport) {
+            this._transport = transport;
+        } else if (!noDefaultTransport) {
+            // Set the default transport.
+            this._transport = new PostMessageTransport(defaultTransportOptions);
```

Comment:
I'm not sure about the idea with passing the specific transport to the constructor. See https://github.com/jitsi/jitsi-meet/pull/1510#issuecomment-296810379

Although you can do that even now trough the transport option, but this is not the default behavior.